### PR TITLE
Adding top-level comment and TOC for functions.handler.

### PR DESCRIPTION
**Pull request by egilmorez**
Adding initial changes for us to build out for functions.handler reference.

Staged internally at https://firebase.devsite.corp.google.com/docs/reference/functions/handler_builder_.handlerbuilder?skip_cache=true

Thanks!

### CODE REVIEW

File: src/handler-builder.ts [hunk 1]
```diff
@@ -35,6 +35,14 @@ import * as remoteConfig from './providers/remoteConfig';
 import * as storage from './providers/storage';
 import * as testLab from './providers/testLab';
 
+/**
+ * The `HandlerBuilder` class facilitates the writing of functions by producers
+ * of Firebase Extensions and developers that want to use gcloud CLI or Cloud Console
+ * to deploy their functions.
```
Comment (rachelsaunders) [inline src/handler-builder.ts]: I didn't know that these were also used for gcloud CLI and Cloud Console?

Comment (egilmorez) [reply src/handler-builder.ts]: It was interesting news to me too!  This addition was requested by a knowledgeable SWE in an earlier review.

File: src/handler-builder.ts [hunk 2]
```diff
@@ -35,6 +35,14 @@ import * as remoteConfig from './providers/remoteConfig';
...
+ *
+ * **Do not use `HandlerBuilder` when writing normal functions for deployment via
+ * Cloud Functions for Firebase.** For normal purposes, use `FunctionBuilder`.
```
Comment (rachelsaunders) [inline src/handler-builder.ts]: Do you want to link to this `FunctionBuilder` content?

File: src/handler-builder.ts [hunk 3]
```diff
@@ -35,6 +35,14 @@ import * as remoteConfig from './providers/remoteConfig';
 import * as storage from './providers/storage';
 import * as testLab from './providers/testLab';
 
+/**
+ * The `HandlerBuilder` class facilitates the writing of functions by producers
+ * of Firebase Extensions and developers that want to use gcloud CLI or Cloud Console
```
Comment (rachelsaunders) [inline src/handler-builder.ts]:
```suggestion
 * of Firebase Extensions and developers that want to use the gcloud CLI or Google Cloud console
```

File: src/handler-builder.ts (hunk 2 above)
Comment (egilmorez) [reply src/handler-builder.ts]: Good suggestion, given the potential for confusion here.

Same for your other edits. However, I'm having issues with npm run format after making them, and would rather just merge this with your approval and make the edits. Later.  SG?

I'm honestly unclear about how the formatting is affected, and have been finding npm run format to be very capricious on a day-by-day basis :)

Comment (rachelsaunders) [reply src/handler-builder.ts]: yep, that SG.

Comment (egilmorez) [reply src/handler-builder.ts]: Actually, bizarrely, I tried one more time to make those edits, and now format likes them :)

File: src/handler-builder.ts [hunk 4]
```diff
@@ -35,6 +35,15 @@ import * as remoteConfig from './providers/remoteConfig';
 import * as storage from './providers/storage';
 import * as testLab from './providers/testLab';
 
+/**
+ * The `HandlerBuilder` class facilitates the writing of functions by producers
+ * of Firebase Extensions and developers who want to use the gcloud CLI or
+ * Google Cloud Console to deploy their functions.
+ *
+ * **Do not use `HandlerBuilder` when writing normal functions for deployment via
+ * Cloud Functions for Firebase.** For normal purposes, use
```
Comment (laurenzlong) [inline src/handler-builder.ts]: `deployment via Firebase CLI` is more understandable.

Comment (egilmorez) [reply src/handler-builder.ts]: Done, thanks!

**Comment by laurenzlong** [comment]
@egilmorez feel free to merge this PR now.

**Pull request merged**